linera-service/tests/end_to_end_tests.rs

```diff
@@ -689,6 +689,10 @@ async fn test_wasm_end_to_end_fungible(
             .make_application(&chain1, &application_id)
             .await?,
     );
+
+    // Needed synchronization though removing it does not get error in 100% of cases.
+    node_service1.process_inbox(&chain1).await?;
```

Comment:
Since the node service is already running, it should receive the incoming message automatically; the failures were just because `assert_balances` was called a bit too quickly. Rather than adding these explicit calls, should we change `assert_balances` so that it retries a few times, with sleeps in between, like we do e.g. in `make_application`?

Reply:
I am open to that way of addressing the issue.

At the same time, the "retry a few times" policy is not exactly great. Eventually, it does not work as we have seen for the tests marked as ignored last week. If we could replace all retry loops by "process_inbox" or similar, I would be happy.

Reply:
I'm not sure; automatically processing incoming messages is a core part of the node service's functionality. But I guess we could test that in only once place, and use `process_inbox` everywhere else.